File: app/src/main/java/br/com/onibusapp/onibusapp/utils/StringUtils.java
package br.com.onibusapp.onibusapp.utils;

import android.content.Context;

/**
 * Created by diego on 26/07/2018.
 */

public final class StringUtils {

    private static StringUtils instance;
    private Context context;

    private StringUtils(Context context) {
        this.context = context;
    }

    public static StringUtils getInstance(Context context) {
        if (instance == null) {
            instance = new StringUtils(context);
        }
        return instance;
    }

    public String getString(int string) {
        return context.getString(string);
    }

}

File: app/src/main/java/br/com/onibusapp/onibusapp/utils/StringUtils.kt
package br.com.onibusapp.onibusapp.utils

import android.content.Context
import java.lang.NumberFormatException

/**
 * Created by diego on 26/07/2018.
 */

class StringUtils {

    fun getStringValues(string: Int, context: Context): String = context.getString(string)

    fun toDouble(valor: String): Double {
        try {
            val numero = valor.replace(',', '.')
            return numero.toDouble()
        } catch (e: NumberFormatException) {
            return 0.0
        }

    }

}
